catroid/src/main/java/org/catrobat/catroid/merge/MergeTask.java

```diff
@@ -132,34 +134,41 @@ public boolean mergeProjects(String mergeProjectName) {
 			}
 		}
 
-		if (oldSceneName != null) {
-			Scene scene;
-			scene = secondProject.getDefaultScene();
-			result &= scene.rename(oldSceneName, activity, false);
+		try {
+			StorageHandler.getInstance().saveProject(mergedProject);
+			ProjectManager.getInstance().loadProject(mergedProject.getName(), activity);
+		} catch (Exception e) {
+			Log.e("MergeTask", "Can't save the merged project");
+			handleError();
+			return false;
 		}
 
-		mergedProject = StorageHandler.getInstance().loadProject(mergedProject.getName(), activity);
-		ProjectManager.getInstance().setCurrentProject(mergedProject);
+		File projectCodeFile = new File(Utils.buildPath(Utils.buildProjectPath(mergeProjectName), Constants.PROJECTCODE_NAME));
+		adapter.insert(new ProjectData(mergeProjectName, projectCodeFile.lastModified()), 0);
+		adapter.notifyDataSetChanged();
 
-		if (adapter != null) {
-			File projectCodeFile = new File(Utils.buildPath(Utils.buildProjectPath(mergeProjectName),
-					Constants.PROJECTCODE_NAME));
-			adapter.insert(new ProjectData(mergeProjectName, projectCodeFile.lastModified()), 0);
-			adapter.notifyDataSetChanged();
+		String msg = firstProject.getName() + " " + activity.getString(R.string.merge_info) + " " + secondProject.getName() + "!";
```

Comment:
is this correct for localisation?

e.g. Arabic is right to left.. so probably add placeholder like %FIRST_PROJECT% in the string resource and string replace it here?